app/workers/bgs/submit_form686c_job.rb

```diff
@@ -11,30 +11,35 @@ class Invalid686cClaim < StandardError; end
 
     sidekiq_options retry: false
 
-    def perform(user_uuid, saved_claim_id, vet_info)
-      user = User.find(user_uuid)
-      Rails.logger.info('BGS::SubmitForm686cJob running!', { user_uuid:, saved_claim_id:, icn: user&.icn })
+    # rubocop:disable Metrics/MethodLength
+    def perform(user_uuid, icn, saved_claim_id, vet_info)
+      Rails.logger.info('BGS::SubmitForm686cJob running!', { saved_claim_id:, icn: })
       in_progress_form = InProgressForm.find_by(form_id: FORM_ID, user_uuid:)
       in_progress_copy = in_progress_form_copy(in_progress_form)
       claim_data = valid_claim_data(saved_claim_id, vet_info)
       normalize_names_and_addresses!(claim_data)
-
-      BGS::Form686c.new(user).submit(claim_data)
+      user_struct = generate_user_struct(vet_info['veteran_information'])
+      if Flipper.enabled?(:dependents_enqueue_with_user_struct)
+        BGS::Form686c.new(user_struct).submit(claim_data)
+      else
+        user = User.find(user_uuid)
+        BGS::Form686c.new(user).submit(claim_data)
+      end
```

Comment:
I notice that the `user`/`user_struct` feature toggle covers `BGS::Form686c.new(*).submit(claim_data)`, but not `send_confirmation_email` or `DependentsApplicationFailureMailer#build` below. I think that's probably fine, but it does make me a bit nervous that if your struct doesn't function as expected here, then it might not function as expected elsewhere, in which case your feature toggle wouldn't let you do a full "rollback."